src/tsdb.c

```diff
@@ -45,20 +46,23 @@ void SeriesTrim(Series * series) {
 
     // start iterator from smallest key
     RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(series->chunks, "^", NULL, 0);
-    Chunk *currentChunk;
+//    Chunk *currentChunk;
+    CompressedChunk *currentChunk;
     void *currentKey;
     size_t keyLen;
     timestamp_t minTimestamp = series->lastTimestamp > series->retentionTime ?
     		series->lastTimestamp - series->retentionTime : 0;
 
     while ((currentKey=RedisModule_DictNextC(iter, &keyLen, (void*)&currentChunk)))
     {
-        if (ChunkGetLastTimestamp(currentChunk) < minTimestamp)
+//        if (ChunkGetLastTimestamp(currentChunk) < minTimestamp)
+        if (CChunk_GetLastTimestamp(currentChunk) < minTimestamp)
```

Comment:
@ashtul please use the original Chunk API + add a makefile switch between compressed chunk and non-compressed chunk.

Reply:
@danni-m @gkorland 
I'm already working on the two chunks implementation.

Functions struct will have the function name without the chunk type (ex. AddSample, GetLastTimestamp...) while the functions themself will get a prefix of `Chunk_` or `CompressedC_`.
This will be done at the end after everything works.
Sounds good?